### DISCUSSION TITLE
better Ripple effect

### DISCUSSION THREAD

**Discussion by BanDroid** [beercss/beercss#389] [Ideas]
hi, my first discussion here, checking out beercss and it is awesome, just the ripple effect is kinda off.

i'm suggesting to use `@gabrielfins/ripple-effect`, it is very lightweight and if you guys agree, i will try to write the same thing in beercss for having the same ripple effect behavior.

**Comment by BanDroid** [comment]
i know it will require a javascript.

maybe we could make beercss dissable the default ripple effect through css and user could add their own ripple? so it will be user choice on having what kind of ripple they want.

provide a data attribute to disable ripple, then user could use that `@gabrielfins/ripple-effect` on their own and add the `md-ripples` to any elements they want

**Reply by BanDroid** [reply]
beercss also use javascript, so i dont know what the downside of this

**Reply by laurentpayot** [reply]
> beercss also use javascript, so i dont know what the downside of this

At least three downsides :wink: 
- It adds a dependency 
- It adds [5KB of JS](https://bundlephobia.com/package/@gabrielfins/ripple-effect) to the bundle
- More JS being run means slower pages display on lower-end machines

**Comment by laurentpayot** [comment]
Why should we integrate `@gabrielfins/ripple-effect` to Beercss by default? Can’t we just have a way for interested users to add it?

**Reply by BanDroid** [reply]
that's what i'm starting the discussion, add a option to disable ripple effect and add another dependency our own, but for now adding another dependency for ripple effect but still have the beercss ripple effect animated looks off

**Reply by Jopie01** [reply]
Take a look at #368. There is code and several examples even without an external library.

**Reply by BanDroid** [reply]
> Take a look at #368. There is code and several examples even without an external library.

dont worry, i'll try to help debugging this

**Comment by leonardorafael** [comment]
Hi,

We have some problems to adopt the majority of ripple effects because:

- it requires JS. And the JS file of Beer CSS was thinked to be optional (not required) working like a helper
- the majority JS solutions of ripple effects, creates a new element inside a container element. The container element needs a overflow to prevent the wave to go outside of the container element dimensions. This will break other elements of Beer CSS (like menu, tooltip, progress...) that needs to go outside of container element dimensions.
- we are carefully trying to get a minimum size on final package (this was a feedback from community)

**Reply by BanDroid** [reply]
`@gabrielfins/ripple-effect` only 2kb

**Reply by leonardorafael** [reply]
Hi @BanDroid ,

It's a very nice effect no doubt. This solution is like other implementations of ripple effect that I saw. It creates a element inside another and uses overflow. There are other solutions with just few lines of code too. But all of them has the same idea, and will break another important elements of Beer CSS like menu, tooltips, progress,... basically all elements that you could use inside another element. I think this is the big problem.

A required JS is not ok for us right now, we have a concept and to break this concept just to have another ripple effect maybe is too much. And finally the size of package, is a constant battle for us, but the community loves it.

It's difficult to have another ripple effect that fits in our project friendly.

**Reply by BanDroid** [reply]
what about making default beercss ripple effect could be disabled using css classes? so this ripple effect could be optional for using different package

**Reply by leonardorafael** [reply]
Try the `no-wave` helper

**Comment by w3teal** [comment]
The whole dependency on a simple CSS framework is too much. And it's also hard to customize and change to fit with Beer.css. I'm already making a simple script for the ripple that can easily customized and fix the `{ overflow: hidden }` issue. Give it a try, https://codepen.io/LIGMATV/pen/mybPNPY.

**Reply by laurentpayot** [reply]
@LIGMATV nice ripple effect with minimal code, I like it :+1:

**Reply by leonardorafael** [reply]
Nice @LIGMATV , will be in the next release. And this resolves the overflow problem, kudos 🍻

**Reply by BanDroid** [reply]
i thought beercss wouldn't accept any js solutions, but here we are 👏

**Reply by BanDroid** [reply]
altough, the codepen solution is not fully handle other events, maybe this one would work better

https://github.com/gabrielfins/ripple-effect/blob/master/src%2Fripples.js

**Comment by leonardorafael** [comment]
Available in [v3.8.0](https://github.com/beercss/beercss/releases/tag/v3.8.0)
